File: FEBRUARY/26-02-2026/Isomorphic Strings.cpp
class Solution {
  public:
    bool areIsomorphic(string &s1, string &s2) {
        // code here
        int n = s1.length();
        vector<int> f1(26, -1), f2(26, 0);
        for(int i = 0; i < n; i++) {
            if(f1[s1[i] - 'a'] == -1 && f2[s2[i] - 'a'] == 0) f1[s1[i] - 'a'] = (s2[i] - 'a');
            else if(f1[s1[i] - 'a'] != s2[i] - 'a') return false;
            f2[s2[i] -'a'] = 1;
        }
        return true;
    }
};

File: FEBRUARY/26-02-2026/Isomorphic Strings.py
class Solution:
     def areIsomorphic(self, s1, s2):
          # code here
          st1 = {}
          st2 = {}

          n = len(s1)
          for i in range(n):
               if s1[i] in st1 or s2[i] in st2:
                    if st1.get(s1[i]) != s2[i] or st2.get(s2[i]) != s1[i]:
                         return False
               else:
                    st1[s1[i]] = s2[i]
                    st2[s2[i]] = s1[i]
          return True